src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,795 @@
+			TreeViewItemSource item5 = new TreeViewItemSource() { Content = "item5" };
+
+			newItems.Add(item4);
+			newItems.Add(item5);
+
+			items.ReplaceAll(newItems);
+
+			Verify.AreEqual(treeView.RootNodes.Count, 2);
+			Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task TreeViewUpdateTest()
+		{
+			TreeView treeView = null;
+			TreeViewNode treeViewNode1 = null;
+			TreeViewNode treeViewNode2 = null;
+			TreeViewNode treeViewNode3 = null;
+			TreeViewNode treeViewNode4 = null;
+			TreeViewNode treeViewNode5 = null;
+
+			await RunOnUIThread.Execute(() =>
+			{
+				treeViewNode1 = new TreeViewNode();
+				treeViewNode2 = new TreeViewNode();
+				treeViewNode3 = new TreeViewNode();
+				treeViewNode4 = new TreeViewNode();
+				treeViewNode5 = new TreeViewNode();
+
+				treeViewNode1.Children.Add(treeViewNode2);
+				treeViewNode1.Children.Add(treeViewNode3);
+				treeViewNode1.Children.Add(treeViewNode4);
+				treeViewNode1.Children.Add(treeViewNode5);
+
+				treeView = new TreeView();
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+
+				var listControl = FindVisualChildByName(treeView, "ListControl") as TreeViewList;
+				treeView.RootNodes.Add(treeViewNode1);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 1);
+
+				treeView.Expand(treeViewNode1);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 5);
+
+				treeViewNode1.Children.RemoveAt(2);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 4);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+				Verify.AreEqual(listControl.GetItemFromIndex(1), treeViewNode2);
+				Verify.AreEqual(listControl.GetItemFromIndex(2), treeViewNode3);
+				Verify.AreEqual(listControl.GetItemFromIndex(3), treeViewNode5);
+
+				treeViewNode1.Children.Insert(1, treeViewNode4);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 5);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+				Verify.AreEqual(listControl.GetItemFromIndex(1), treeViewNode2);
+				Verify.AreEqual(listControl.GetItemFromIndex(2), treeViewNode4);
+				Verify.AreEqual(listControl.GetItemFromIndex(3), treeViewNode3);
+				Verify.AreEqual(listControl.GetItemFromIndex(4), treeViewNode5);
+
+				treeViewNode1.Children.Clear();
+				Verify.AreEqual(listControl.GetItems()?.Count(), 1);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+		}
+
+		//[TestMethod] Disabled with issue number #1775 (WinUI issue)
+		public void TreeViewInheritanceTest()
+		{
+			StackPanel stackPanel = new StackPanel();
+			SpecialTreeView inheritedTreeView = new SpecialTreeView();
+			SpecialTreeViewList inheritedTreeViewList = new SpecialTreeViewList();
+			SpecialTreeViewItem inheritedTreeViewItem = new SpecialTreeViewItem();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'inheritedTreeViewItem'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)